methods/guest/src/bin/is_even.rs

```diff
@@ -14,34 +14,22 @@
 
 #![no_main]
 
-use std::io::Read;
-
-use ethabi::{ethereum_types::U256, ParamType, Token};
+use alloy_primitives::U256;
+use alloy_sol_types::SolValue;
 use risc0_zkvm::guest::env;
 
 risc0_zkvm::guest::entry!(main);
 
-fn fibonacci(n: U256) -> U256 {
-    let (mut prev, mut curr) = (U256::one(), U256::one());
-    for _ in 2..=n.as_u32() {
-        (prev, curr) = (curr, prev + curr);
-    }
-    curr
-}
-
-fn main() {
+pub fn main() {
     // Read data sent from the application contract.
-    let mut input_bytes = Vec::<u8>::new();
-    env::stdin().read_to_end(&mut input_bytes).unwrap();
-    // Type array passed to `ethabi::decode_whole` should match the types encoded in
-    // the application contract.
-    let input = ethabi::decode_whole(&[ParamType::Uint(256)], &input_bytes).unwrap();
-    let n: U256 = input[0].clone().into_uint().unwrap();
+    let mut number: U256 = env::read();
 
     // Run the computation.
-    let result = fibonacci(n);
+    if number.bit(0) {
+        number = U256::ZERO;
+    }
```

Comment:
I actually think we should frame this as a check, and panic on failure, instead of returning zero. Returning zero does indeed work, but it's not obvious why it works on first read, and it leads to weird behavior in error cases. E.g. if the user runs the example commands and gives `5`, the proof will succeed, they will post a transaction to the chain, and then when they run `get` it will return 0. They will simply think something is broken.
```suggestion
assert!(number.bit(0) == 0);
```